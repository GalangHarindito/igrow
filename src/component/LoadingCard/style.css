.card-loading {
  display: flex;
  justify-content: center;

}

.card-loading-container {
  background: #fff;
  background: linear-gradient(110deg, #ececec 8%, #f5f5f5 18%, #ececec 33%);
  width: 18rem;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  border-radius: 2rem;
  box-shadow: 0px 1rem 1.5rem rgb(0 0 0 / 50%);
  margin-right: 2rem;
  animation: 1.5s shine linear infinite;
}

@keyframes shine {
  10% {
    opacity: 1;
    top: -30%;
    left: -30%;
    transition-property: left, top, opacity;
    transition-duration: 0.2s, 0.2s, 0.5s;
    transition-timing-function: ease;
  }
  100% {
    opacity: 0.4;
    top: -30%;
    left: -30%;
    transition-property: left, top, opacity;
  }
}

.card-loading-container:last-child {
  margin-right: 0rem;
  
}

.card-loading-image {
  width: 10rem;
  height: 10rem;
  background-repeat: no-repeat;
  background-size: cover;
  border: 0.5rem solid #eee;
  border-radius: 50%;
  -webkit-border-radius: 50%;
  -moz-border-radius: 50%;
  -ms-border-radius: 50%;
  -o-border-radius: 50%;
  
}

.card-loading-container > div:nth-child(1) {
  background-color: #eee;
  padding: 1rem;
  background-size: cover;
}

.card-loading-container > div:nth-child(1) > div {
  background-color: #fff;
  padding: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translate(30%, 50%);
  -moz-transform: translate(30%, 50%);
  -ms-transform: translate(30%, 50%);
  -o-transform: translate(30%, 50%);
  -webkit-transform: translate(30%, 50%);
}
.card-loading-container > div:nth-child(2) {
  height: 3.5rem;
}

.card-loading-container > div:nth-child(3) {
  padding: 1.2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.card-loading-container > div:nth-child(3) > p {
  padding: 0.2rem 0;
}

.name-loading {
  background-color: #fff;
  height: 1.3rem;
  width: 10rem;
  padding: 0.rem;
  margin-bottom: 0.7rem;
}
.location-loading {
  background-color: #fff;
  height: 1.3rem;
  width: 10rem;
  
}

.phone-loading {
  background-color: #fff;
  height: 1.3rem;
  width: 10rem;
  margin-bottom: 0.7rem;
}

@media screen and (max-width: 700px) {
  .card-loading-container {
    margin-right: 0rem;
    margin-bottom: 2rem;
  }

  
}