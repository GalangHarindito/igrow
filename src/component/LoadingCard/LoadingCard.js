import React from "react";
import "./style.css";

export default function LoadingCard(props) {
  const { data } = props;

  return (
    <section className='card-loading'>
      {data.map((el, idx) => {
        return (
          <section className='card-loading-container' key={idx}>
            <div>
              <div className='card-loading-image'></div>
            </div>
            <div></div>
            <div>
              <div className='name-loading'></div>
              <div className='phone-loading'></div>
              <div className='location-loading'></div>
            </div>
          </section>
        );
      })}
    </section>
  );
}
