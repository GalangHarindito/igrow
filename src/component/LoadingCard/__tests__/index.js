import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoadingCard from '../LoadingCard';


describe('src/component/Card', () => {
  const data = [
    {
      name : 'test',
      gender: 'test',
      phone: 'test',
      picture: { large:'test' },
      location: { city: 'test', country: 'test'}
    },
    {
      name : 'test',
      gender: 'test',
      phone: 'test',
      picture: { large:'test' },
      location: { city: 'test', country: 'test'}
    }
  ];

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<LoadingCard data={data} />);
    expect(tree).toMatchSnapshot();
  });

});