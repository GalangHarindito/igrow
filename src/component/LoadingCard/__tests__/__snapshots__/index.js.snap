// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/Card render 1`] = `
<section
  className="card-loading"
>
  <section
    className="card-loading-container"
  >
    <div>
      <div
        className="card-loading-image"
      />
    </div>
    <div />
    <div>
      <div
        className="name-loading"
      />
      <div
        className="phone-loading"
      />
      <div
        className="location-loading"
      />
    </div>
  </section>
  <section
    className="card-loading-container"
  >
    <div>
      <div
        className="card-loading-image"
      />
    </div>
    <div />
    <div>
      <div
        className="name-loading"
      />
      <div
        className="phone-loading"
      />
      <div
        className="location-loading"
      />
    </div>
  </section>
</section>
`;
